### ISSUE TITLE
error using load_lora_weights with dora

### ISSUE THREAD

**Issue by SlZeroth** [huggingface/peft#1561]
pipe = StableDiffusionXLPipeline.from_pretrained(
    "stabilityai/stable-diffusion-xl-base-1.0",
    torch_dtype=torch.bfloat16
)

pipe.to("cuda")

pipe.load_lora_weights("e:/diffusion-models/Lora/test16/woman_young_64rank.safetensors")
pipe.fuse_lora(lora_scale=0.85)

got error. this is error information

<img width="1230" alt="image" src="https://github.com/huggingface/peft/assets/8098724/255dcdda-54c2-459f-aa0e-6e71af6b18d6">

**Comment by SlZeroth** [comment]
I tested with main branch peft code and diffusers version is 0.25.0

**Comment by BenjaminBossan** [comment]
Instead of posting a screenshot, could you please copy the entire error message as text and paste it here?

**Comment by SlZeroth** [comment]
this is error log 

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 2
      1 # 1131144.safetensors
----> 2 pipe.load_lora_weights("e:/diffusion-models/Lora/test16/woman_young_64rank.safetensors")
      3 pipe.fuse_lora(lora_scale=0.85)

File ~\anaconda3\envs\runpod-dev\lib\site-packages\diffusers\loaders\lora.py:1442, in StableDiffusionXLLoraLoaderMixin.load_lora_weights(self, pretrained_model_name_or_path_or_dict, adapter_name, **kwargs)
   1414 """
   1415 Load LoRA weights specified in `pretrained_model_name_or_path_or_dict` into `self.unet` and
   1416 `self.text_encoder`.
   (...)
   1435         See [`~loaders.LoraLoaderMixin.lora_state_dict`].
   1436 """
   1437 # We could have accessed the unet config from `lora_state_dict()` too. We pass
   1438 # it here explicitly to be able to tell that it's coming from an SDXL
   1439 # pipeline.
   1440 
   1441 # First, ensure that the checkpoint is a compatible one and can be successfully loaded.
-> 1442 state_dict, network_alphas = self.lora_state_dict(
   1443     pretrained_model_name_or_path_or_dict,
   1444     unet_config=self.unet.config,
   1445     **kwargs,
   1446 )
   1447 is_correct_format = all("lora" in key for key in state_dict.keys())
   1448 if not is_correct_format:

File ~\anaconda3\envs\runpod-dev\lib\site-packages\huggingface_hub\utils\_validators.py:118, in validate_hf_hub_args.<locals>._inner_fn(*args, **kwargs)
    115 if check_use_auth_token:
    116     kwargs = smoothly_deprecate_use_auth_token(fn_name=fn.__name__, has_token=has_token, kwargs=kwargs)
--> 118 return fn(*args, **kwargs)

File ~\anaconda3\envs\runpod-dev\lib\site-packages\diffusers\loaders\lora.py:299, in LoraLoaderMixin.lora_state_dict(cls, pretrained_model_name_or_path_or_dict, **kwargs)
    296     if unet_config is not None:
    297         # use unet config to remap block numbers
    298         state_dict = _maybe_map_sgm_blocks_to_diffusers(state_dict, unet_config)
--> 299     state_dict, network_alphas = _convert_kohya_lora_to_diffusers(state_dict)
    301 return state_dict, network_alphas

File ~\anaconda3\envs\runpod-dev\lib\site-packages\diffusers\loaders\lora_conversion_utils.py:270, in _convert_kohya_lora_to_diffusers(state_dict, unet_name, text_encoder_name)
    267         network_alphas.update({new_name: alpha})
    269 if len(state_dict) > 0:
--> 270     raise ValueError(f"The following keys have not been correctly be renamed: \n\n {', '.join(state_dict.keys())}")
    272 logger.info("Kohya-style checkpoint detected.")
    273 unet_state_dict = {f"{unet_name}.{module_name}": params for module_name, params in unet_state_dict.items()}

ValueError: The following keys have not been correctly be renamed: 

 lora_te1_text_model_encoder_layers_0_mlp_fc1.dora_scale, lora_te1_text_model_encoder_layers_0_mlp_fc2.dora_scale, lora_te1_text_model_encoder_layers_0_self_attn_k_proj.dora_scale, lora_te1_text_model_encoder_layers_0_self_attn_out_proj.dora_scale, lora_te1_text_model_encoder_layers_0_self_attn_q_proj.dora_scale, lora_te1_text_model_encoder_layers_0_self_attn_v_proj.dora_scale, lora_te1_text_model_encoder_layers_10_mlp_fc1.dora_scale, lora_te1_text_model_encoder_layers_10_mlp_fc2.dora_scale, lora_te1_text_model_encoder_layers_10_self_attn_k_proj.dora_scale, lora_te1_text_model_encoder_layers_10_self_attn_out_proj.dora_scale, lora_te1_text_model_encoder_layers_10_self_attn_q_proj.dora_scale, lora_te1_text_model_encoder_layers_10_self_attn_v_proj.dora_scale, lora_te1_text_model_encoder_layers_11_mlp_fc1.dora_scale, lora_te1_text_model_encoder_layers_11_mlp_fc2.dora_scale, lora_te1_text_model_encoder_layers_11_self_attn_k_proj.dora_scale, lora_te1_text_model_encoder_layers_11_self_attn_out_proj.dora_scale, lora_te1_text_model_encoder_layers_11_self_attn_q_proj.dora_scale, lora_te1_text_model_encoder_layers_11_self_attn_v_proj.dora_scale, lora_te1_text_model_encoder_layers_1_mlp_fc1.dora_scale, lora_te1_text_model_encoder_layers_1_mlp_fc2.dora_scale, lora_te1_text_model_encoder_layers_1_self_attn_k_proj.dora_scale, lora_te1_text_model_encoder_layers_1_self_attn_out_proj.dora_scale, lora_te1_text_model_encoder_layers_1_self_attn_q_proj.dora_scale, lora_te1_text_model_encoder_layers_1_self_attn_v_proj.dora_scale, lora_te1_text_model_encoder_layers_2_mlp_fc1.dora_scale, lora_te1_text_model_encoder_layers_2_mlp_fc2.dora_scale, lora_te1_text_model_encoder_layers_2_self_attn_k_proj.dora_scale,

**Comment by BenjaminBossan** [comment]
Indeed, it looks like `_convert_kohya_lora_to_diffusers` does not take into account the newly added DoRA parameters. However, I'm not sure where `dora_scale` is coming from, in PEFT the extra parameter is called `lora_magnitude_vector`. @SlZeroth do you know the origin of this adapter?

Pinging @sayakpaul in case he has an idea.

**Comment by SlZeroth** [comment]
https://drive.google.com/file/d/1NhfYy1tJXxbPWUXetiM0SQC1L-60f3sk/view?usp=sharing

okay thank you., this is my dora file. can you test with it?

**Comment by SlZeroth** [comment]
https://huggingface.co/streamize/test-dora/tree/main

I uploaded lora file to huggingface thank you!

**Comment by sayakpaul** [comment]
Thank you! Just for curiosity, how's the visual quality of the LoRA checkpoint?

**Comment by SlZeroth** [comment]
https://ibb.co/FqckVRd this is the LORA checkpoint image
